rust/connlib/tunnel/src/client.rs

```diff
@@ -1183,6 +1181,66 @@ mod tests {
         )
     }
 
+    #[test]
+    fn add_resources_works() {
+        let mut client_state = ClientState::for_test();
+        client_state.add_resources(&[cidr_resource("10.0.0.0/24"), dns_resource("baz.com")]);
+
+        assert_eq!(
+            HashSet::<&ResourceDescription>::from_iter(client_state.resources().iter()),
+            HashSet::from_iter([cidr_resource("10.0.0.0/24"), dns_resource("baz.com")].iter())
+        );
+        assert_eq!(
+            HashSet::<IpNetwork>::from_iter(client_state.routes()),
+            expected_routes(vec![IpNetwork::from_str("10.0.0.0/24").unwrap()])
+        );
```

Comment:
I think "having routes for CIDR resources" should be a separate test.

Reply:
I think for now, it's a good idea to leave them here until we have better tests for routes in general, we could repeat all tests with these assertions.

But for now, it's the only way we have to know that the internal state is consistent, it has catched some bugs.